Update openms_ci_matrix_full.yml

poshul opened the pull request:
Reenable rsync in openms_ci_matrix_full

poshul committed:
Update openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

try to upload documentation

jpfeuffer commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -613,4 +650,4 @@ jobs:
           echo "$PASS" > ~/.ssh/private.key
           sudo chmod 600 ~/.ssh/private.key
-#          rsync --progress -avz -e "ssh -i ~/.ssh/private.key -o StrictHostKeyChecking=no" $GITHUB_WORKSPACE/de.openms.update/target/repository/* "$USER@$HOST:/knime-plugin/updateSite/$folder/"
+          rsync --progress -avz -e "ssh -i ~/.ssh/private.key -o StrictHostKeyChecking=no" $GITHUB_WORKSPACE/de.openms.update/target/repository/* "$USER@$HOST:/knime-plugin/updateSite/$folder/"
```
Replicate the above "GHA" folder logics here, if you are not shutting down Jenkins?

jpfeuffer commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -413,7 +421,35 @@ jobs:
+          else
+            folder=experimental/${{ env.RUN_NAME }}
+          fi
+          echo $folder
```
Remove or make it useful ? "Uploading to archive folder $f"

jpfeuffer reviewed (approved):
Overall looks good.
You could add the NSIS changes that @cbielow proposed if he is not yet working on it.

poshul committed:
Update openms_ci_matrix_full.yml

Fixes from suggestions

poshul commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -361,6 +361,15 @@ jobs:
           ${{ github.workspace }}/OpenMS/bld/*.deb
           ${{ github.workspace }}/OpenMS/bld/*.pkg
+
+    - if: steps.set-vars.outputs.pkg_type != 'none' && [[ "${{ matrix.os }}" == ubuntu-* ]] # Only upload docs when we are building the package, use the ubuntu build simply 'cause its fast
```
@jpfeuffer this seems to trigger on both Ubuntu runners, unlike uploading the installers. Am I missing something in the syntax of the if statement?

jpfeuffer replied:
Yep. See the suggestion that we merged. You don't need the square brackets in if fields.

poshul replied:
Thanks, I saw that your update fixed it. I'm currently trying to figure out why the html/ directory doesn't build.

jpfeuffer commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -361,6 +361,15 @@ jobs:
           ${{ github.workspace }}/OpenMS/bld/*.deb
           ${{ github.workspace }}/OpenMS/bld/*.pkg
+
+    - if: steps.set-vars.outputs.pkg_type != 'none' && [[ "${{ matrix.os }}" == ubuntu-* ]] # Only upload docs when we are building the package, use the ubuntu build simply 'cause its fast
```
```suggestion
    - if: steps.set-vars.outputs.pkg_type != 'none' && matrix.os == ubuntu-* 
    # Only upload docs when we are building the package, use the ubuntu build simply 'cause its fast
```

jpfeuffer replied:
Maybe you also need quotes around ubuntu-*

poshul committed:
Merge branch 'develop' into GHA-full-fixes

poshul committed:
Update .github/workflows/openms_ci_matrix_full.yml

Co-authored-by: Julianus Pfeuffer <j.pfeuffer@fu-berlin.de>

poshul committed:
Update openms_ci_matrix_full.yml

jpfeuffer commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -361,6 +361,16 @@ jobs:
           ${{ github.workspace }}/OpenMS/bld/*.deb
           ${{ github.workspace }}/OpenMS/bld/*.pkg
+
+    - if: steps.set-vars.outputs.pkg_type != 'none' && matrix.os == "ubuntu-*" 
```
```suggestion
    - if: steps.set-vars.outputs.pkg_type != 'none' && matrix.os == 'ubuntu-*'
```

jpfeuffer committed:
Update .github/workflows/openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

Enable tutorial building update path to download docs artifiact

poshul committed:
Update openms_ci_matrix_full.yml

revert tutorial change

jpfeuffer committed:
test doxygen

jpfeuffer commented:
You probably need to upload the docs from the build directory.

poshul commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -156,6 +156,8 @@ jobs:
           if [[ "${{ steps.set-vars.outputs.pkg_type }}" != "none" ]]; then
             sudo apt-get -qq install -y doxygen ghostscript graphviz
+            echo "Testing doxygen"
+            doxygen --version
```
This seems to (somehow) make the Mac runner also install Doxygen (which then fails). I cannot for the life of me figure out how, since this is in a block the only gets run on Ubuntu.

jpfeuffer replied:
I think macOS has its own doxygen installation command which gets run. If you don't want to install it on mac, you should remove it from there.

jpfeuffer replied:
Doxygen was installed all the time. There was just an update to one of the brew formulas that broke everything. I expect it to break in the nightlies today as well.

poshul committed:
Disable building docs on windows and Mac (temporarily)

poshul committed:
Merge branch 'develop' into GHA-full-fixes

jpfeuffer commented on .github/workflows/openms_ci_matrix_full.yml:
```diff
@@ -361,6 +367,16 @@ jobs:
+      with:
+        name: documentation
+        path: |
+          ${{ github.workspace }}/OpenMS/doc/**
```
```suggestion
          ${{ github.workspace }}/OpenMS/bld/doc/**
```

jpfeuffer replied:
This is what I mean basically

jpfeuffer replied:
Maybe use a var if we have any for the build dir

poshul committed:
Merge branch 'develop' into GHA-full-fixes

jpfeuffer commented:
Docs are required for the installers. Why disable?

poshul committed:
Update .github/workflows/openms_ci_matrix_full.yml

Co-authored-by: Julianus Pfeuffer <j.pfeuffer@fu-berlin.de>

poshul commented:
> Docs are required for the installers. Why disable?

I chatted with Timo yesterday about whether we should be building them on all the nightlies, I wanted to see how broken the installers were without them (in the case of future issues like what's happening right now with macos)

poshul committed:
Update openms_ci_matrix_full.yml

Reenable docs on windows

poshul committed:
Update openms_ci_matrix_full.yml

Zip documentation before uploading as artifact.

jpfeuffer commented:
Since the macOS installer is completely CMake-based it should work without docs. It is solely depending on which CMake components are built and installed (in CMake lingo).

The problem is that for Windows we have a custom NSIS script that we have to adapt everytime we make changes in our CMake code (unless we completely templatize the script by adding even more CMake configure template variables).
A hack would be to make the "File" install calls in the NSIS script optional but you will not be able to do that for the install sections easily.

Unfortunately the default CMake-based NSIS generator does not support complex things like user/system installs etc. but it can be combined with a template for the non component-related things with some work.

poshul commented:
> Since the macOS installer is completely CMake-based it should work without docs. It is solely depending on which CMake components are built and installed (in CMake lingo).
> 
> The problem is that for Windows we have a custom NSIS script that we have to adapt everytime we make changes in our CMake code (unless we completely templatize the script by adding even more CMake configure template variables). A hack would be to make the "File" install calls in the NSIS script optional but you will not be able to do that for the install sections easily.
> 
> Unfortunately the default CMake-based NSIS generator does not support complex things like user/system installs etc. but it can be combined with a template for the non component-related things with some work.

So I took a quick look at this this morning, and it looks like there is an option in our NSIS template to disable inclusion of docs: https://github.com/OpenMS/OpenMS/blob/d313ff5c77c101c091b383971381576d312447b4/cmake/Windows/NSIS.template.in#L33

I've not really looked into it any further

poshul committed:
Update openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

poshul committed:
Update openms_ci_matrix_full.yml

use directory instead of path on zip action

jpfeuffer commented:
I would omit the CMakeFiles subdirectories.
Or just upload index.html and html.
Or CMake install the documentation to another folder and zip from there.

poshul commented:
> I would omit the CMakeFiles subdirectories. Or just upload index.html and html. Or CMake install the documentation to another folder and zip from there.

Okay. We currently upload the whole thing, so I wasn't sure if there was another purpose for it that I'm missing.

jpfeuffer commented:
I'm not sure how I did it in Jenkins. Maybe we have a cmake mechanism that copies/moves everything relevant away from the build folder.
But the only thing I can think of is the CMake install that by default should put it into a share directory.

poshul committed:
Update openms_ci_matrix_full.yml

Exclude non-html stuff

poshul committed:
Update openms_ci_matrix_full.yml

Fix base directory for Rsync

poshul committed:
Update openms_ci_matrix_full.yml

Fix trailing slash

poshul commented:
@jpfeuffer this look good? I've still got the mac doc building off since it's still broken.

jpfeuffer commented:
I need to check the artifacts but I think the code looks ok.
Thanks for the comment on the zip. What was the error with the "::"? Is there an issue for it.

MacOS brew is also annoying. Hundreds of people have the same problem though https://github.com/actions/runner-images/issues/11545

I bet there is a workaround somewhere in the dozens of linked issues.

jpfeuffer commented:
People usually seem to do a brew install python@3.11 --force --overwrite in the beginning.

poshul commented:
![Screenshot 2023-11-18 100402](https://github.com/OpenMS/OpenMS/assets/146827/f254db3f-b8e0-49c3-9f1f-3815957572f3)
Here's the output regarding the `::` problem. I didn't open a bug for it since it appears to be intentional (here's someone else requesting documentation on it https://github.com/actions/upload-artifact/issues/333).

I'm gonna open another PR to fix the nightlies with your suggestion.

poshul committed:
Update openms_ci_matrix_full.yml

Renable docs for Mac when building package, disable building docs when no package.

poshul committed:
Merge branch 'develop' into GHA-full-fixes
